teps/0091-trusted-resources.md

```diff
@@ -0,0 +1,329 @@
+(i.e., `...@sha256:abcdef`) in order for the verification to succeed.
+In later iterations, this can be extended such that tag-referenced images can
+_themselves_ be verified for the overall Task to pass verification. However,
+this will be left out of the scope of this TEP.
+
+The Pipeline Author will be able to provide optional configuration to Tekton
+Chains such that Chains will *only* consider results from Tasks that were
+verifiable (and, following the previous rule, were successfully verified).
+
+
+### Risks and Mitigations
+
+* Mutating webhook is a risk of the proposal. The proposed verification will happen after the mutating webhook, so the content of a Resource may be mutated and fail the verification. This would be an issue for local cluster resources.
+
+  Some possible mitigations include the following:
+
+  1. Verify the Tasks/Pipelines when applying them in the cluster, and update the Resources to add the missing mutating values (e.g. Use `SetDefaults`) everytime we bump the version of Tekton Pipeline.
+  2. Avoid mutating Tekton Resources or move the mutating after verification.
+
+* When verifying a Remote Resource that identifies a Task, the Task still has
+references to external, unverified resources - namely,
+the builder images. These builder images may have been compromised and the
+Task would still be considered verified.
+
+  Some possible mitigations include the following:
+
+  1. (Within scope of this proposal) For a Task to be considered verified, all
+    builder image references within it are suggested to be by-digest. The digest references
+    would form a low-effort way for the Task Author to signal that they have
+    verified or otherwise trust the builder images they are referencing.
+    For example, they may have manually verified the builder images using Cosign
+    or some other technique.
+  2. (Outside the scope of this proposal) Configuration is introduced into Tekton
+    to facilitate verification of the builder images themselves. Configuration
+    would also be required to cover off various combinations of bundle
+    verification and image verification - for example, do I allow verified
+    Task Bundles if the Task within the Bundle does not have verified images?
+
+### User Experience (optional)
+
+**Author publishing a Resource.**
+A Resource Author may want their Resource and its
+outputs to be trusted so that provenance attestations can be confidently made. The author will use the private key to sign the Resource and signatures are stored in `Annotations` map of the Resource. Public keys are stored as secrets or via URI stored in the configmap. Similar mechanism is used in [Chains](https://github.com/tektoncd/chains).
+
+**Verify the Resource via validating webhook.**
+The user will need to deploy the validating webhook to the cluster, by default the webhook will reject a Resource if it fails the verification. The webhook will mark if this Resource fails or passes the verification.
```

Comment:
> Will the deployment be on by default and enforcement controlled via configuration?
This will be gated by alpha and a new configmap, and by default it should just skip the verification. This section is not updated and I will update right now!